benchmarks/ImageProcessing/BuddyCorr2DBenchmark.cpp

```diff
@@ -70,9 +83,17 @@ void initializeBuddyCorr2D(char **argv) {
 
   sizesOutputBuddyCorr2D[0] = outputRowsBuddyCorr2D;
   sizesOutputBuddyCorr2D[1] = outputColsBuddyCorr2D;
+
+  string argv3 = argv[3];
```

Comment:
Why capture the CLI argument in a separate dedicated variable?

Reply:
Because we cannot directly compare `char*` using `==` operator. [Reference](https://cplusplus.com/forum/beginner/47996/#msg260512).
Instead using dedicated variable maybe I can use 
```
if(string(argv[3]) == "REPLICATE_PADDING")
```

Reply:
Using a separate variable for the purpose of typecasting seems unnecessary to me. 
You have two options to solve this problem, 
1. Use `C` utilities such as `strcmp(argv[3], "REPLICATE_PADDING")`.
2. Use typecasting : 
At this stage, I would prefer the `C++` way of dealing with typecasting (which provides more safety than standard `C` style casts -> [Ref](https://stackoverflow.com/a/1255015)). With this, you can simply try `if (static_cast<string>(argv[3]) == "REPLICATE_PADDING")`.